### PR TITLE
AbsLine survey advances

### PR DESCRIPTION
**Pull request by profxj**
Bits and pieces throughout the modules in linetools.isgm
Mainly for writing and loading those objects.

Also, a few updates related to limits in components.

A new test.  
No new docs aside from doc strings.

### CODE REVIEW

File: linetools/guis/xspecgui.py [hunk 1]
```diff
@@ -83,9 +83,11 @@ def __init__(self, ispec, parent=None, zsys=None, norm=None, exten=None,
         if hasattr(self.spec_widg.spec, 'stypes'):
             if self.spec_widg.spec.stypes[self.spec_widg.select].lower() == 'galaxy':
                 self.pltline_widg.llist = ltgu.set_llist('Galaxy',in_dict=self.pltline_widg.llist)
-                self.pltline_widg.llist['Plot'] = True
-                idx = self.pltline_widg.lists.index(self.pltline_widg.llist['List'])
-                self.pltline_widg.llist_widget.setCurrentRow(idx)
+            elif self.spec_widg.spec.stypes[self.spec_widg.select].lower() == 'absorber':
```
[the remark this answers is not in the document]
Comment (profxj) [reply linetools/guis/xspecgui.py]: only in the CASBaH repo

File: linetools/isgm/abscomponent.py [hunk 2]
```diff
@@ -433,7 +438,7 @@ def add_absline(self, absline, tol=0.1*u.arcsec, chk_vel=True,
              (allows for round-off error)
         chk_sep : bool, optional
           Perform coordinate check (expensive)
-        vtoler : float
+        vtoler : float, optional
```
[the remark this answers is not in the document]
Comment (profxj) [reply linetools/isgm/abscomponent.py]: yeah, let's not

File: linetools/isgm/abscomponent.py [hunk 3]
```diff
@@ -679,22 +684,20 @@ def plot_Na(self, show=True, **kwargs):
...
         for aline in self._abslines:
-            if aline.analy['vlim'][0] < self.vlim[0]:
-                if verbose:
-                    print('Resetting vlim0 from {}'.format(aline))
-                self.vlim[0] = aline.analy['vlim'][0]
-            if aline.analy['vlim'][1] > self.vlim[1]:
-                if verbose:
-                    print('Resetting vlim1 from {}'.format(aline))
-                self.vlim[1] = aline.analy['vlim'][1]
+            zmin = min(zmin, aline.limits.zlim[0])
```
[the remark this answers is not in the document]
Comment (profxj) [reply linetools/isgm/abscomponent.py]: take another look.  
Am pretty sure it is right.

File: linetools/isgm/abssystem.py [hunk 4]
```diff
@@ -627,31 +629,31 @@ def to_dict(self):
...
 
     def update_vlim(self, sub_system=None):
-        """ Update vlim in the main or subsystems using the components
+        """ Update zLimits in the main or subsystems using the components
 
         Parameters
         ----------
         sub_system : str, optional
           If provided, apply to given sub-system.  Only used in LLS so far
+          Not working anymore
```
[the remark this answers is not in the document]
Comment (profxj) [reply linetools/isgm/abssystem.py]: It is only for LLS.  
Has been semi-depracated for awhile.

**Comment by profxj** [comment]
Check my responses @jnburchett

File: linetools/isgm/abscomponent.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (jnburchett) [reply linetools/isgm/abscomponent.py]: Oh yeah...I get it now.

File: linetools/isgm/abssystem.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (jnburchett) [reply linetools/isgm/abssystem.py]: Oh yeah, I see the comment applies only to this parameter....

**Pull request merged**